walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,135 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.*; 
+import java.util.Comparator;
+import java.util.Collections;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    
+    List<TimeRange> mandatoryAttendeeAvailability = new ArrayList<>();
+    List<TimeRange> allAttendeeAvailability = new ArrayList<>();
+
+    //if the meeting is scheduled for longer than a day, it cannot be scheduled
+    if (request.getDuration() > 24*60) {
+      return allAttendeeAvailability;
+    } 
+
+    //if no attendees or no meetings, any time of day works
+    else if ((request.getAttendees().size() == 0 && request.getOptionalAttendees().size() == 0) || events.size() == 0) {
+      allAttendeeAvailability.add(TimeRange.WHOLE_DAY);
+      return allAttendeeAvailability;
+    } 
+
+    //otherwise, find all possible meeting times   
+    else {
```

Comment:
Same comment about the closing bracket + `else`

You'd also usually not use a newline at the start or end of a code block (much like for function definitions), ie.

```
if (foo) {
   code.starts("here");
}
```

as opposed to

```
if (bar) {

  not.likeThis();
}
```